### PR TITLE
Add non-default noiseShape support to dropout op

### PR DESCRIPTION
**Pull request by syt123450**
This PR is a followup PR for [tfjs-core#1343](https://github.com/tensorflow/tfjs-core/pull/1343) which added `dropout` op, dropout feature initially requested in [tfjs#117](https://github.com/tensorflow/tfjs/issues/117). This PR cleans up a TODO, `implements non default noise shape`. The implementation of noiseShape aligns with TensorFlow Python's dropout API.

The `noiseShape` feature would benefit further dropout related development, for example, make `tf.layers.dropout` support `noiseShape` configuration.

**Relative PR:**
* Make dropout layer support non default noiseShape and seed [tensorflow/tfjs-layers#556](https://github.com/tensorflow/tfjs-layers/pull/556)

**Reference:**
* [TensorFlow tf.nn.dropout documentation](https://www.tensorflow.org/api_docs/python/tf/nn/dropout)
* [TensorFlow tf.nn.dropout implementation](https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/python/ops/nn_ops.py#L2982-L3054)
* [TensorFlow _get_noise_shape implementation](https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/python/ops/nn_ops.py#L2903-L2925)

---
To see the logs from the Cloud Build CI, please join either
our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs)
or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/1782)

### CODE REVIEW

**Comment by caisq** [comment]
Just as a heads-up: tfjs-layers PR https://github.com/tensorflow/tfjs-layers/pull/556 is blocked on this one.
